docs/Building_Diagrams/service_tasks.md

```diff
@@ -0,0 +1,98 @@
+
+- **Python Expressions**: All fields in SpiffWorkflow interpret input as Python expressions. Wrap url in single quotes (`'...'`) unless passing them as variables.
+- **Authentication**: For APIs requiring Basic Auth, use `basic_auth_username` and `basic_auth_password` parameters.
+- **Parameter Syntax**: Headers, query parameters, and URLs require JSON-like syntax compatible with Python dictionaries.
+- **Response Variables**: Use response variables to store data retrieved from service tasks. These variables can be referenced in later tasks for display or further processing.
+
+Below, we’ll walk through detailed setup instructions for two examples to illustrate different configurations and use cases.
+
+### Example 1: Retrieving Employee Information from BambooHR API
+In this example, we’ll configure a Service Task that fetches employee details from the BambooHR API and displays this data in a subsequent manual task. 
+
+Below is workflow overview:
+
+![Service Task](images/service_task_doc1.png)
+
+1. **Start Event**: Begins the workflow.
+2. **Service Task**: Retrieves employee information from BambooHR.
+3. **Manual Task**: Displays the retrieved employee data to the user.
+4. **End Event**: Completes the workflow.
+
+#### **Service Task Configuration**
+  ```{image} ./images/service_task_doc2.png
+:alt: Login Screen
+:class: bg-primary mb-1
+:width: 300px
+:align: right
+```
+- **Task Name**: `Get Employee Info` 
+- **Operator ID**: `http/GetRequest`
+  - This operator is used for making HTTP GET requests to retrieve data from an external API.
+- **Response Variable**: `bamboo_get_employee`
+  - This variable stores the API response, allowing it to be used in subsequent tasks.
+  
+
+
+**Parameters**
+1. **Basic Authentication**: 
+   - **basic_auth_password**: `"x"` (enclosed in quotes as its a python expression).
+   - **basic_auth_username**: `"secret:BAMBOOHR_API_KEY"`
+     - Replace `BAMBOOHR_API_KEY` with your actual API key. This format specifies the username for Basic Auth, using secure handling with `secret:` notation.
+
+2. **Headers**:
+   - **headers**: `{"Accept": "application/json"}`
+     - Specifies that the response should be JSON formatted.
+
+3. **Query Parameters**:
+   - **params**: `{"fields": "firstName,lastName"}`
+     - Defines specific fields (e.g., `firstName`, `lastName`) to retrieve from the API.
+
+4. **URL**:
+   - **url**: `'https://api.bamboohr.com/api/gateway.php/{your_company_subdomain}/v1/employees/directory'`
+     - Replace `{your_company_subdomain}` with your BambooHR subdomain.
+
+### Example 2: Fetching Mock Data from JSONPlaceholder API
+
+This simpler example demonstrates how to retrieve mock data from the JSONPlaceholder API, useful for testing or prototyping workflows.
+
+#### Service Task Configuration
+  ```{image} ./images/service_task_doc3.png
+:alt: Service Task
+:width: 300px
+:align: right
+```
+- **Task Name**: `Get Post Data`
+- **Operator ID**: `http/GetRequestV2`
```

Comment:
We should probably use get request V2 for the other one as well

Reply:
Or maybe this one should actually use an http post to create an album or something so both are not http get?